common/core/desktop/src/kmx/kmx_xstring.cpp

```diff
@@ -116,37 +117,37 @@ km_kbp_cp *km::kbp::kmx::u16tok(km_kbp_cp *p, km_kbp_cp ch, km_kbp_cp **ctx) {
 *
 * xstrlen calculates the length of a string, ignoring some special chars.
 */
+PKMX_WCHAR km::kbp::kmx::incxstr(PKMX_WCHAR p) {
 
-PKMX_WCHAR km::kbp::kmx::incxstr(PKMX_WCHAR p)
-{
-  if(*p == 0) return p;
-  if(*p != UC_SENTINEL)
-  {
-    if(*p >= 0xD800 && *p <= 0xDBFF && *(p+1) >= 0xDC00 && *(p+1) <= 0xDFFF) return p+2;
-    return p+1;
+PKMX_WCHAR km::kbp::kmx::incxstr(PKMX_WCHAR p) {
```

Comment:
This code doesn't compile -- can you make sure it compiles and then I will review?

Reply:
changed this file, included your proposals